### PR TITLE
Add ObjectNet task

### PR DESCRIPTION
**Pull request by kotekjedi**
ObjectNet task implementation.

### CODE REVIEW

File: shifthappens/tasks/objectnet/objectnet.py [hunk 1]
```diff
@@ -0,0 +1,112 @@
...
+from shifthappens import benchmark as sh_benchmark
+from shifthappens.data.base import DataLoader
+from shifthappens.models import base as sh_models
+from shifthappens.models.base import PredictionTargets
+from shifthappens.tasks.base import Task
+from shifthappens.tasks.metrics import Metric
+from shifthappens.tasks.objectnet import objectnet_utils
+from shifthappens.tasks.task_result import TaskResult
+
+JSON_URL = "https://raw.githubusercontent.com/abarbu/objectnet-template-tensorflow/master/mapping_files/imagenet_id_to_objectnet_id.json"
```
Comment (zimmerrol) [inline shifthappens/tasks/objectnet/objectnet.py:35]: Let's move this also to the class definition, like the `resources` field.

File: shifthappens/tasks/objectnet/objectnet.py [hunk 2]
```diff
@@ -0,0 +1,112 @@
...
+
+        if not os.path.exists(json_path):
+            mapping_json = pd.read_json(JSON_URL, typ="series")
+            mapping_json.to_json(json_path)
+        # ImageNet -> ObjectNet mapping
+        self.mapping_json = pd.read_json(json_path, typ="series").to_dict()
+
+        test_transform = tv_transforms.Compose(
+            [
+                tv_transforms.Pad(-2),
```
Comment (zimmerrol) [inline shifthappens/tasks/objectnet/objectnet.py:76]: Why do we apply this padding here?

File: shifthappens/tasks/objectnet/objectnet_utils.py [hunk 3]
```diff
@@ -0,0 +1,138 @@
...
+import urllib.error
+import zipfile
+from typing import Optional
+
+import numpy as np
+import pandas as pd
+from torchvision import datasets
+
+
+# udapted from https://github.com/pytorch/vision/blob/0cba9b7845795d6be7b164037461ea6e9265f6a2/torchvision/datasets/utils.py
```
Comment (zimmerrol) [inline shifthappens/tasks/objectnet/objectnet_utils.py:13]: typo

File: shifthappens/tasks/objectnet/objectnet_utils.py [hunk 4]
```diff
@@ -0,0 +1,138 @@
...
+        k: class_to_imagenet[v]
+        for k, v in class_to_folder_name.items()
+        if v in class_to_imagenet
+    }
+    return mapping
+
+
+class ImageFolderImageNetClassesIntersection(datasets.ImageFolder):
+    """
+    This is required for handling empty folders that cannot be mapped classes in ImageNet.
```
Comment (zimmerrol) [inline shifthappens/tasks/objectnet/objectnet_utils.py:85]: typo: cannot be mapped to classes

File: shifthappens/tasks/objectnet/objectnet_utils.py [hunk 5]
```diff
@@ -0,0 +1,138 @@
...
+    """
+
+    def __init__(self, mapping_path, *args, **kwargs):
+        self.folder_map = folder_map(mapping_path)
+        super().__init__(*args, **kwargs)
+
+    def find_classes(self, directory):
+        """Rejects the folders that cannot be mapped classes in ImageNet."""
```
Comment (zimmerrol) [inline shifthappens/tasks/objectnet/objectnet_utils.py:93]: typo: cannot be mapped to

**Review by zimmerrol — CHANGES_REQUESTED** [review]
Small changes requested, but overall the code looks good! Can you please also add a CITATION.cff (see https://github.com/shift-happens-benchmark/icml-2022/pull/60/files#diff-cff7599d4e8605d087d90de757b2c4c93644510cd58ad270c070ce65ce1e2cfa for an example)

File: shifthappens/tasks/objectnet/objectnet.py (hunk 2 above)
Comment (kotekjedi) [reply shifthappens/tasks/objectnet/objectnet.py:76]: Every ObjectNet image has a tiny red border across it. This is written on the ObjectNet website "We provide ObjectNet with a 1 pixel red border around each image which must be removed before performing inference. The ObjectNet license requires that if you post images from ObjectNet to the web, you include this border. Any time you see an image with a solid 1 pixel red border, that's an indication it's in someone's test set and you should be careful about training on it. "

Although the authors stated that it is only a 1-pixel border I visualized the transformed image with a `tv_transforms.Pad(-1)`, and it still has this border, so I use `tv_transforms.Pad(-2)` here.

File: shifthappens/tasks/objectnet/objectnet.py [hunk 6]
```diff
@@ -0,0 +1,122 @@
...
+    """
+    The class that wraps evaluation of models' robustness on the ObjectNet
+    subset intersecting with ImageNet classes.
+    """
+
+    imagenet_to_objectnet_json_url = (
+        "https://raw.githubusercontent.com"
+        "/abarbu/objectnet-template-tensorflow"
+        "/master/mapping_files"
+        "/imagenet_id_to_objectnet_id.json "
```
Comment (kotekjedi) [inline shifthappens/tasks/objectnet/objectnet.py:50]:
```suggestion
        "/imagenet_id_to_objectnet_id.json"
```

File: shifthappens/tasks/objectnet/objectnet_utils.py [hunk 7]
```diff
@@ -0,0 +1,139 @@
...
+        k: class_to_imagenet[v]
+        for k, v in class_to_folder_name.items()
+        if v in class_to_imagenet
+    }
+    return mapping
+
+
+class ImageFolderImageNetClassesIntersection(datasets.ImageFolder):
+    """
+    This is required for handling empty folders that cannot be mapped to
```
Comment (kotekjedi) [inline shifthappens/tasks/objectnet/objectnet_utils.py:85]:
```suggestion
    This is required for handling folders that cannot be mapped to
```

**Review by zimmerrol — APPROVED** [review]

**Pull request merged**